### ISSUE TITLE
Linux JVM crash with netty 4.1.60 in io.netty.channel.epoll.LinuxSocket.newSocketStream

### ISSUE THREAD

**Issue by sfc-gh-pbennes** [netty/netty#11149]
At this point I don't have a minimal reproduction case I can provide. Any advice you can give me to build one would be helpful. I've got a monolithic application that started crashing attempting to update io.netty past 4.1.59. https://github.com/netty/netty/commit/56adab2743e90175c958c1fb0ce8857a851e02ea possibly related?

### Expected behavior
No JVM crash

### Actual behavior
JVM crash

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f84b90842f9, pid=25029, tid=28386
#
# JRE version: OpenJDK Runtime Environment (12.0.2+10) (build 12.0.2+10)
# Java VM: OpenJDK 64-Bit Server VM (12.0.2+10, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  0x00007f84b90842f9
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/libexec/abrt-hook-ccpp %s %c %p %u %g %t e %P %I %h" (or dumping to /home/pbennes/app/core.25029)
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-build/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  S U M M A R Y ------------

Command Line: -Djava.net.preferIPv4Stack=true -Dio.netty.tryReflectionSetAccessible=true --add-opens=jdk.management/com.sun.management.internal=ALL-UNNAMED --add-exports=java.base/com.sun.net.ssl.internal.ssl=ALL-UNNAMED --add-exports=java.base/sun.util.calendar=ALL-UNNAMED -Djava.security.egd=file:/dev/urandom ...

Host: Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz, 7 cores, 19G, CentOS Linux release 7.9.2009 (Core)
Time: Thu Apr  8 16:10:02 2021 UTC elapsed time: 148 seconds (0d 0h 2m 28s)

---------------  T H R E A D  ---------------

Current thread (0x00007f8788468000):  JavaThread "parallel-1" daemon [_thread_in_native, id=28386, stack(0x00007f84ba4bb000,0x00007f84bb8bc000)]

Stack: [0x00007f84ba4bb000,0x00007f84bb8bc000],  sp=0x00007f84bb8b9308,  free space=20472k
Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  0x00007f84b90842f9
j  io.netty.channel.unix.Socket.newSocketStream0(Z)I+1
j  io.netty.channel.epoll.LinuxSocket.newSocketStream(Z)Lio/netty/channel/epoll/LinuxSocket;+5
j  io.netty.channel.epoll.LinuxSocket.newSocketStream()Lio/netty/channel/epoll/LinuxSocket;+3
j  io.netty.channel.epoll.EpollSocketChannel.<init>()V+1
j  reactor.netty.resources.DefaultLoopEpoll.getChannel(Ljava/lang/Class;)Lio/netty/channel/Channel;+13
j  reactor.netty.resources.LoopResources.onChannel(Ljava/lang/Class;Lio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+24
j  reactor.netty.tcp.TcpResources.onChannel(Ljava/lang/Class;Lio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+20
j  reactor.netty.transport.TransportConfig.lambda$connectionFactory$0(ZLio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+10
j  reactor.netty.transport.TransportConfig$$Lambda$3596.newChannel()Lio/netty/channel/Channel;+12
j  reactor.netty.transport.TransportConnector.doInitAndRegister(Lreactor/netty/transport/TransportConfig;Lio/netty/channel/ChannelInitializer;Z)Lreactor/core/publisher/Mono;+18
j  reactor.netty.transport.TransportConnector.connect(Lreactor/netty/transport/TransportConfig;Ljava/net/SocketAddress;Lio/netty/resolver/AddressResolverGroup;Lio/netty/channel/ChannelInitializer;)Lreactor/core/publisher/Mono;+34
j  reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator.lambda$connectChannel$0(Lreactor/core/publisher/MonoSink;)V+23
j  reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$$Lambda$3585.accept(Ljava/lang/Object;)V+8
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.core.publisher.Mono.subscribe(Lorg/reactivestreams/Subscriber;)V+71
j  reactor.core.publisher.Mono.subscribeWith(Lorg/reactivestreams/Subscriber;)Lorg/reactivestreams/Subscriber;+2
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;Ljava/lang/Runnable;Lreactor/util/context/Context;)Lreactor/core/Disposable;+14
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;Ljava/lang/Runnable;)Lreactor/core/Disposable;+8
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;)Lreactor/core/Disposable;+12
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.drainLoop()V+234
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.drain()V+11
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.doAcquire(Lreactor/netty/internal/shaded/reactor/pool/AbstractPool$Borrower;)V+26
j  reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.request(J)V+97
j  reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onSubscribe(Lorg/reactivestreams/Subscription;)V+52
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool$QueueBorrowerMono.subscribe(Lreactor/core/CoreSubscriber;)V+26
j  reactor.netty.resources.PooledConnectionProvider.lambda$acquire$1(Ljava/util/function/Supplier;Lreactor/netty/transport/TransportConfig;Lio/netty/resolver/AddressResolverGroup;Lreactor/netty/ConnectionObserver;Lreactor/core/publisher/MonoSink;)V+102
j  reactor.netty.resources.PooledConnectionProvider$$Lambda$3579.accept(Ljava/lang/Object;)V+24
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect.lambda$subscribe$0(Lreactor/netty/http/client/HttpClientConnect$HttpClientHandler;Lreactor/core/publisher/MonoSink;)V+286
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$$Lambda$3544.accept(Ljava/lang/Object;)V+12
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.core.publisher.FluxRetryWhen.subscribe(Lreactor/core/CoreSubscriber;Lreactor/util/retry/Retry;Lreactor/core/CorePublisher;)V+106
j  reactor.core.publisher.MonoRetryWhen.subscribeOrReturn(Lreactor/core/CoreSubscriber;)Lreactor/core/CoreSubscriber;+9
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+4
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.MonoDefer.subscribe(Lreactor/core/CoreSubscriber;)V+39
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.MonoDelaySubscription.accept(Lreactor/core/publisher/FluxDelaySubscription$DelaySubscriptionOtherSubscriber;)V+16
j  reactor.core.publisher.MonoDelaySubscription.accept(Ljava/lang/Object;)V+5
j  reactor.core.publisher.FluxDelaySubscription$DelaySubscriptionOtherSubscriber.onNext(Ljava/lang/Object;)V+27
j  reactor.core.publisher.MonoDelay$MonoDelayRunnable.run()V+31
j  reactor.core.scheduler.SchedulerTask.call()Ljava/lang/Void;+49
j  reactor.core.scheduler.SchedulerTask.call()Ljava/lang/Object;+1
J 14034 c1 java.util.concurrent.FutureTask.run()V java.base@12.0.2 (123 bytes) @ 0x00007f888e1cdd6c [0x00007f888e1cdba0+0x00000000000001cc]
J 14884 c2 java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run()V java.base@12.0.2 (57 bytes) @ 0x00007f88946d177c [0x00007f88946d1520+0x000000000000025c]
J 21633 c2 java.util.concurrent.ThreadPoolExecutor.runWorker(Ljava/util/concurrent/ThreadPoolExecutor$Worker;)V java.base@12.0.2 (187 bytes) @ 0x00007f8894a44514 [0x00007f8894a44260+0x00000000000002b4]
J 7985 c1 java.util.concurrent.ThreadPoolExecutor$Worker.run()V java.base@12.0.2 (9 bytes) @ 0x00007f888d81aedc [0x00007f888d81ae40+0x000000000000009c]
J 7979 c1 java.lang.Thread.run()V java.base@12.0.2 (17 bytes) @ 0x00007f888d818844 [0x00007f888d8186e0+0x0000000000000164]
v  ~StubRoutines::call_stub
V  [libjvm.so+0x8961e9]  JavaCalls::call_helper(JavaValue*, methodHandle const&, JavaCallArguments*, Thread*)+0x3c9
V  [libjvm.so+0x89417d]  JavaCalls::call_virtual(JavaValue*, Handle, Klass*, Symbol*, Symbol*, Thread*)+0x1ed
V  [libjvm.so+0x94465c]  thread_entry(JavaThread*, Thread*)+0x6c
V  [libjvm.so+0xf1b9cf]  JavaThread::run()+0x23f
V  [libjvm.so+0xf185ef]  Thread::call_run()+0x16f
V  [libjvm.so+0xc2e0ce]  thread_native_entry(Thread*)+0xee

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  io.netty.channel.unix.Socket.newSocketStreamFd(Z)I+0
j  io.netty.channel.unix.Socket.newSocketStream0(Z)I+1
j  io.netty.channel.epoll.LinuxSocket.newSocketStream(Z)Lio/netty/channel/epoll/LinuxSocket;+5
j  io.netty.channel.epoll.LinuxSocket.newSocketStream()Lio/netty/channel/epoll/LinuxSocket;+3
j  io.netty.channel.epoll.EpollSocketChannel.<init>()V+1
j  reactor.netty.resources.DefaultLoopEpoll.getChannel(Ljava/lang/Class;)Lio/netty/channel/Channel;+13
j  reactor.netty.resources.LoopResources.onChannel(Ljava/lang/Class;Lio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+24
j  reactor.netty.tcp.TcpResources.onChannel(Ljava/lang/Class;Lio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+20
j  reactor.netty.transport.TransportConfig.lambda$connectionFactory$0(ZLio/netty/channel/EventLoopGroup;)Lio/netty/channel/Channel;+10
j  reactor.netty.transport.TransportConfig$$Lambda$3596.newChannel()Lio/netty/channel/Channel;+12
j  reactor.netty.transport.TransportConnector.doInitAndRegister(Lreactor/netty/transport/TransportConfig;Lio/netty/channel/ChannelInitializer;Z)Lreactor/core/publisher/Mono;+18
j  reactor.netty.transport.TransportConnector.connect(Lreactor/netty/transport/TransportConfig;Ljava/net/SocketAddress;Lio/netty/resolver/AddressResolverGroup;Lio/netty/channel/ChannelInitializer;)Lreactor/core/publisher/Mono;+34
j  reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator.lambda$connectChannel$0(Lreactor/core/publisher/MonoSink;)V+23
j  reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$$Lambda$3585.accept(Ljava/lang/Object;)V+8
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.core.publisher.Mono.subscribe(Lorg/reactivestreams/Subscriber;)V+71
j  reactor.core.publisher.Mono.subscribeWith(Lorg/reactivestreams/Subscriber;)Lorg/reactivestreams/Subscriber;+2
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;Ljava/lang/Runnable;Lreactor/util/context/Context;)Lreactor/core/Disposable;+14
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;Ljava/lang/Runnable;)Lreactor/core/Disposable;+8
j  reactor.core.publisher.Mono.subscribe(Ljava/util/function/Consumer;Ljava/util/function/Consumer;)Lreactor/core/Disposable;+12
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.drainLoop()V+234
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.drain()V+11
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.doAcquire(Lreactor/netty/internal/shaded/reactor/pool/AbstractPool$Borrower;)V+26
j  reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.request(J)V+97
j  reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onSubscribe(Lorg/reactivestreams/Subscription;)V+52
j  reactor.netty.internal.shaded.reactor.pool.SimpleDequePool$QueueBorrowerMono.subscribe(Lreactor/core/CoreSubscriber;)V+26
j  reactor.netty.resources.PooledConnectionProvider.lambda$acquire$1(Ljava/util/function/Supplier;Lreactor/netty/transport/TransportConfig;Lio/netty/resolver/AddressResolverGroup;Lreactor/netty/ConnectionObserver;Lreactor/core/publisher/MonoSink;)V+102
j  reactor.netty.resources.PooledConnectionProvider$$Lambda$3579.accept(Ljava/lang/Object;)V+24
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect.lambda$subscribe$0(Lreactor/netty/http/client/HttpClientConnect$HttpClientHandler;Lreactor/core/publisher/MonoSink;)V+286
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$$Lambda$3544.accept(Ljava/lang/Object;)V+12
j  reactor.core.publisher.MonoCreate.subscribe(Lreactor/core/CoreSubscriber;)V+21
j  reactor.core.publisher.FluxRetryWhen.subscribe(Lreactor/core/CoreSubscriber;Lreactor/util/retry/Retry;Lreactor/core/CorePublisher;)V+106
j  reactor.core.publisher.MonoRetryWhen.subscribeOrReturn(Lreactor/core/CoreSubscriber;)Lreactor/core/CoreSubscriber;+9
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+4
j  reactor.netty.http.client.HttpClientConnect$MonoHttpConnect.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.MonoDefer.subscribe(Lreactor/core/CoreSubscriber;)V+39
j  reactor.core.publisher.InternalMonoOperator.subscribe(Lreactor/core/CoreSubscriber;)V+33
j  reactor.core.publisher.MonoDelaySubscription.accept(Lreactor/core/publisher/FluxDelaySubscription$DelaySubscriptionOtherSubscriber;)V+16
j  reactor.core.publisher.MonoDelaySubscription.accept(Ljava/lang/Object;)V+5
j  reactor.core.publisher.FluxDelaySubscription$DelaySubscriptionOtherSubscriber.onNext(Ljava/lang/Object;)V+27
j  reactor.core.publisher.MonoDelay$MonoDelayRunnable.run()V+31
j  reactor.core.scheduler.SchedulerTask.call()Ljava/lang/Void;+49
j  reactor.core.scheduler.SchedulerTask.call()Ljava/lang/Object;+1
J 14034 c1 java.util.concurrent.FutureTask.run()V java.base@12.0.2 (123 bytes) @ 0x00007f888e1cdd6c [0x00007f888e1cdba0+0x00000000000001cc]
J 14884 c2 java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run()V java.base@12.0.2 (57 bytes) @ 0x00007f88946d177c [0x00007f88946d1520+0x000000000000025c]
J 21633 c2 java.util.concurrent.ThreadPoolExecutor.runWorker(Ljava/util/concurrent/ThreadPoolExecutor$Worker;)V java.base@12.0.2 (187 bytes) @ 0x00007f8894a44514 [0x00007f8894a44260+0x00000000000002b4]
J 7985 c1 java.util.concurrent.ThreadPoolExecutor$Worker.run()V java.base@12.0.2 (9 bytes) @ 0x00007f888d81aedc [0x00007f888d81ae40+0x000000000000009c]
J 7979 c1 java.lang.Thread.run()V java.base@12.0.2 (17 bytes) @ 0x00007f888d818844 [0x00007f888d8186e0+0x0000000000000164]
v  ~StubRoutines::call_stub

siginfo: si_signo: 11 (SIGSEGV), si_code: 128 (SI_KERNEL), si_addr: 0x0000000000000000

Register to memory mapping:

RAX=0x00000000b8e924f4 is an unallocated location in the heap
RBX={method} {0x00007f877c740af0} 'newSocketStreamFd' '(Z)I' in 'io/netty/channel/unix/Socket'
RCX=0x0000000000000020 is an unknown value
RDX=0x0 is NULL
RSP=0x00007f84bb8b9308 is pointing into the stack for thread: 0x00007f8788468000
RBP=0x00007f84bb8b9368 is pointing into the stack for thread: 0x00007f8788468000
RSI=0x00007f84bb8b9378 is pointing into the stack for thread: 0x00007f8788468000
RDI=0x00007f8788468330 points into unknown readable memory: 80 25 69 ab 88 7f 00 00
R8 ={method} {0x00007f877c7b65d8} '<init>' '()V' in 'io/netty/channel/epoll/EpollSocketChannel'
R9 =0x0000000000000004 is an unknown value
R10=0x00007f888c498941 is at code_begin+897 in an Interpreter codelet
method entry point (kind = native)  [0x00007f888c4985c0, 0x00007f888c499000]  2624 bytes
R11=0x00000000a71ee0b8 points into unknown readable memory: 31 67 20 e8 47 00 00 00
R12=0x0 is NULL
R13={method} {0x00007f877c740af0} 'newSocketStreamFd' '(Z)I' in 'io/netty/channel/unix/Socket'
R14=0x00007f84bb8b9388 is pointing into the stack for thread: 0x00007f8788468000
R15=0x00007f8788468000 is a thread
```

```
Internal exceptions (10 events):
Event: 148.340 Thread 0x00007f87f803f800 Exception <a 'com/apple/foundationdb/FDBException'{0x00000000f4a30738}> (0x00000000f4a30738) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/prims/jni.cpp, line 615]
Event: 148.397 Thread 0x00007f8788468000 Exception <a 'java/lang/UnsatisfiedLinkError'{0x00000000f4213b68}: io.netty.incubator.channel.uring.Native.createFile()I> (0x00000000f4213b68) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/ho
Event: 148.397 Thread 0x00007f8788468000 Exception <a 'java/lang/reflect/InvocationTargetException'{0x00000000f4215b58}> (0x00000000f4215b58) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/runtime/reflection.cpp, line 1
Event: 148.399 Thread 0x00007f8788468000 Exception <a 'java/lang/NoSuchMethodError'{0x00000000f4232230}: Method io.netty.channel.unix.Socket.sendTo(IZLjava/nio/ByteBuffer;II[BII)I not found> (0x00000000f4232230) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspo
Event: 148.399 Thread 0x00007f8788468000 Exception <a 'java/lang/NoSuchMethodError'{0x00000000f4232230}> (0x00000000f4232230) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/prims/jni.cpp, line 615]
Event: 148.400 Thread 0x00007f8788468000 Exception <a 'java/lang/reflect/InvocationTargetException'{0x00000000f42327a8}> (0x00000000f42327a8) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/runtime/reflection.cpp, line 1
Event: 148.400 Thread 0x00007f8788468000 Exception <a 'java/lang/NoSuchMethodError'{0x00000000f4233530}: Method io.netty.channel.unix.Socket.sendTo(IZLjava/nio/ByteBuffer;II[BII)I not found> (0x00000000f4233530) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspo
Event: 148.400 Thread 0x00007f8788468000 Exception <a 'java/lang/NoSuchMethodError'{0x00000000f4233530}> (0x00000000f4233530) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/prims/jni.cpp, line 615]
Event: 148.401 Thread 0x00007f8788468000 Exception <a 'java/lang/LinkageError'{0x00000000f4233c00}> (0x00000000f4233c00) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/oops/instanceKlass.cpp, line 1053]
Event: 148.425 Thread 0x00007f87f81a7800 Exception <a 'com/apple/foundationdb/FDBException'{0x00000000f46ef590}> (0x00000000f46ef590) thrown at [/home/jenkins/workspace/build-scripts/jobs/jdk12u/jdk12u-linux-x64-hotspot/workspace/build/src/src/hotspot/share/prims/jni.cpp, line 615]
```

### Steps to reproduce
TBD

### Minimal yet complete reproducer code (or URL to code)
TBD

### Netty version
4.1.60

### JVM version (e.g. `java -version`)
AdoptOpenJDK 12 HotSpot. Also tested with 16 and can still reproduce.
`OpenJDK 64-Bit Server VM AdoptOpenJDK (build 12.0.2+10, mixed mode, sharing)`

### OS version (e.g. `uname -a`)
CentOS 7. `Linux DEVVM-pbennes 3.10.0-1160.21.1.el7.x86_64 #1 SMP Tue Mar 16 18:28:22 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by sfc-gh-pbennes** [comment]
This appears to be the result of some dependency graph interaction. I seem to have stopped the crash by updating a few other dependencies that rely on io.netty: com.linecorp.armeria:armeria 1.3.0 -> 1.6.0, io.projectreactor.netty:reactor-netty-core 1.0.4 -> 1.0.5. If I can figure out specifically what the bad interaction is I'll report back..

**Comment by chrisvest** [comment]
@sfc-gh-pbennes Any chance your previous set of dependencies pulled in multiple different Netty versions?

Notably, 4.1.59 introduce new protections to prevent native libraries from being loaded multiple times: https://github.com/netty/netty/commit/5b41f3d25b9709b9f4c779938aa7a4b6550b727f – if you have pre-59 and 59+ versions on the class path at the same time, I could imagine this might break in weird ways.

**Comment by sfc-gh-pbennes** [comment]
We're using io.grpc:grpc-netty-shaded which looks like it shades netty 4.1.51.Final, which I could see causing a conflict. It was one of the dependencies I updated that stopped the conflict though that shades 4.1.52.Final which doesn't seem like much of an update. Today I'm going to try to nail down what is causing the conflict and that seems like a plausible place to start.

**Comment by sfc-gh-pbennes** [comment]
It looks like updating com.linecorp.armeria:armeria 1.3.0 -> 1.4.0 fixes my crash.

1.3.0 depends on 4.1.54 and 1.4.0 on 4.1.58. I see https://github.com/line/armeria/pull/3244 which..could maybe explain the issue? I don't feel like git bisecting armeria because testing my monolith app for this crash takes ~14 minutes per attempt.

Since my immediate issue is resolved by updating armeria to 1.4.0 we can close this issue and if nothing else it can serve as a reference if anyone else encounters this issue. IDK if there's anything to-do here otherwise.
